src/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeFileUpload.tsx

```diff
@@ -66,10 +82,89 @@ export const DefendDisputeFileUpload = () => {
         ];
     }, [goBack, defendDisputeCallback, i18n, isFetching]);
 
+    const [oneOrMoreSelectedDocument, setOneOrMoreSelectedDocument] = useState<string | undefined>(undefined);
+    const [optionalSelectedDocuments, setOptionalSelectedDocuments] = useState<(string | undefined)[]>([]);
+
+    const { requiredDocuments, optionalDocuments, oneOrMoreDocuments } = useMemo(() => {
+        const docs: { requiredDocuments: string[]; optionalDocuments: SelectItem<string>[]; oneOrMoreDocuments: SelectItem<string>[] } = {
+            requiredDocuments: [],
+            optionalDocuments: [],
+            oneOrMoreDocuments: [],
+        };
+
+        (applicableDocuments ?? []).forEach(({ documentTypeCode, requirementLevel }) => {
+            const name = getDefenseDocumentContent(i18n, documentTypeCode)?.title || documentTypeCode;
+            switch (requirementLevel) {
+                case 'REQUIRED':
+                    docs.requiredDocuments.push(documentTypeCode);
+                    break;
+                case 'OPTIONAL':
+                    docs.optionalDocuments.push({
+                        id: documentTypeCode,
+                        name,
+                    });
+                    break;
+                case 'ONE_OR_MORE':
+                    docs.oneOrMoreDocuments.push({
+                        id: documentTypeCode,
+                        name,
+                    });
+                    break;
+            }
+            return docs;
+        });
+
+        return docs;
+    }, [applicableDocuments, i18n]);
+
+    const addOptionalDocument = useCallback((documentType?: string, index?: number) => {
+        if (documentType === undefined) {
+            setOptionalSelectedDocuments(prev => [...prev, documentType]);
+        } else if (index !== undefined) {
+            setOptionalSelectedDocuments(prev => {
+                if (prev[index] === documentType) {
+                    return prev;
+                }
+                const newDocs = [...prev];
+                newDocs[index] = documentType;
+                return newDocs;
+            });
+        }
+    }, []);
+
+    const addEmptyOptionalDocument = useCallback(() => {
+        addOptionalDocument();
+    }, [addOptionalDocument]);
+
+    const availableOptionalDocuments = useMemo(() => {
+        const additionalOptionalDocs = oneOrMoreDocuments.filter(doc => doc.id !== oneOrMoreSelectedDocument);
+
+        return [...additionalOptionalDocs, ...optionalDocuments].map(doc => {
+            return { ...doc, disabled: optionalSelectedDocuments.includes(doc.id) ?? false };
```

Comment:
No need for the _nullish coalescing_ (`??`) operation here. Since the `includes()` method always returns a `boolean`, that branch of the expression is never evaluated.

```suggestion
            return { ...doc, disabled: optionalSelectedDocuments.includes(doc.id) };
```

Reply:
✅